CMakeLists.txt

```diff
@@ -181,6 +181,33 @@ endif()
 
 if(NOT Protobuf_FOUND)
   find_package(Protobuf REQUIRED)
+
+  # orchestrated effort of both protobuf & cmake are responsible for the next lines..
+
+  # TODO: monitor how protobuf Linux packages evolve. Eventually everyone should use
+  # the Apple configuration, but e.g. apt does not install CMake config with protobuf
+  if(Protobuf_VERSION VERSION_GREATER_EQUAL 4.22.0)
+    add_definitions(-DPROTOBUF_USE_DLLS)
```

Comment:
what does this do? the way it reads is that it would try to link dynamically rather than statically. if that is the case, i'd like to say that we shouldnt be making this choice for the person building. on platforms such as android and ios it makes way more sense to statically link so we dont have to worry about what versions of libraries are lying around. is this needed?

Reply:
apparently it's needed for (some) version on apple: https://github.com/apache/arrow/issues/35987. not entirely sure it's really needed, no. IIRC I built without it just fine on M1. I'll comment it out for now with that arrow issue